huaweicloud/services/dli/resource_huaweicloud_dli_flinkjar_job.go

```diff
@@ -0,0 +1,512 @@
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.DliV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DLI v1 client, err=%s", err)
+	}
+
+	aErr := authorizeObsBucket(client, d)
+	if aErr != nil {
+		return aErr
+	}
+
+	opts := flinkjob.CreateJarJobOpts{
+		Name:                 d.Get("name").(string),
+		Desc:                 d.Get("description").(string),
+		QueueName:            d.Get("queue_name").(string),
+		MainClass:            d.Get("main_class").(string),
+		Entrypoint:           d.Get("entrypoint").(string),
+		EntrypointArgs:       d.Get("entrypoint_args").(string),
+		Feature:              d.Get("feature").(string),
+		FlinkVersion:         d.Get("flink_version").(string),
+		Image:                d.Get("image").(string),
+		CuNumber:             golangsdk.IntToPointer(d.Get("cu_number").(int)),
+		ParallelNumber:       golangsdk.IntToPointer(d.Get("parallel_number").(int)),
+		ObsBucket:            d.Get("obs_bucket").(string),
+		LogEnabled:           utils.Bool(d.Get("log_enabled").(bool)),
+		SmnTopic:             d.Get("smn_topic").(string),
+		RestartWhenException: utils.Bool(d.Get("restart_when_exception").(bool)),
+		ManagerCuNumber:      golangsdk.IntToPointer(d.Get("manager_cu_number").(int)),
+		TmCus:                golangsdk.IntToPointer(d.Get("tm_cus").(int)),
+		TmSlotNum:            golangsdk.IntToPointer(d.Get("tm_slot_num").(int)),
+		ResumeCheckpoint:     utils.Bool(d.Get("resume_checkpoint").(bool)),
+		ResumeMaxNum:         golangsdk.IntToPointer(d.Get("resume_max_num").(int)),
+		CheckpointPath:       d.Get("checkpoint_path").(string),
+		Tags:                 utils.ExpandResourceTags(d.Get("tags").(map[string]interface{})),
+	}
+
+	if runtimConfig, ok := d.GetOk("runtime_config"); ok {
+		config := utils.ExpandResourceTags(runtimConfig.(map[string]interface{}))
+		configStr, _ := json.Marshal(config)
+		opts.RuntimeConfig = string(configStr)
+	}
+
+	if dependencyJars, ok := d.GetOk("dependency_jars"); ok {
+		var dependencyArray []string
+		for _, v := range dependencyJars.([]interface{}) {
+			dependencyArray = append(dependencyArray, v.(string))
+		}
+		opts.DependencyJars = dependencyArray
+	}
+
+	if dependencyFiles, ok := d.GetOk("dependency_files"); ok {
+		var dependencyArray []string
+		for _, v := range dependencyFiles.([]interface{}) {
+			dependencyArray = append(dependencyArray, v.(string))
+		}
+		opts.DependencyFiles = dependencyArray
+	}
+
+	logp.Printf("[DEBUG] Creating new DLI flink jar job opts: %#v", opts)
+
+	rst, createErr := flinkjob.CreateJarJob(client, opts)
+	if createErr != nil {
+		return fmtp.DiagErrorf("Error creating DLI flink jar job: %s", createErr)
+	}
+
+	if rst != nil && !rst.IsSuccess {
+		return fmtp.DiagErrorf("Error creating DLI flink jar job: %s", rst.Message)
+	}
+
+	d.SetId(strconv.Itoa(rst.Job.JobId))
```

Comment:
Suggest to update golangsdk to convert the job ID to a string type.

Reply:
keep it as it is.  if use string. maybe pass a string which is not number to sdk